src/misc.ts

```diff
@@ -0,0 +1,92 @@
+//
+// Types
+//
+
+/**
+ * Makes every specified property of the specified object type mutable.
+ *
+ * @template T - The object whose readonly properties to make mutable.
+ * @template TargetKey - The property key(s) to make mutable.
+ */
+export type Mutable<
+  T extends Record<string, unknown>,
+  TargetKey extends string
```

Comment:
```suggestion
  TargetKey extends keyof T
```
Again not sure about the use of `T` in this file.